src/test/javascript/spec/component/programming-assessment/programming-assessment-manual-result.component.spec.ts

```diff
@@ -229,81 +217,80 @@ describe('CodeEditorTutorAssessmentContainerComponent', () => {
+        expect(assessmentLayout).not.toBe(null);
     });
 
     it('should show complaint for result with complaint and check assessor', fakeAsync(() => {
         comp.ngOnInit();
         tick(100);
 
-        expect(getIdentityStub.calledOnce).to.be.true;
-        expect(lockAndGetProgrammingSubmissionParticipationStub.calledOnce).to.be.true;
-        expect(findBySubmissionIdStub.calledOnce).to.be.true;
-        expect(comp.isAssessor).to.be.true;
-        expect(comp.complaint).to.exist;
+        expect(getIdentityStub).toHaveBeenCalledTimes(1);
+        expect(lockAndGetProgrammingSubmissionParticipationStub).toHaveBeenCalledTimes(1);
+        expect(findBySubmissionIdStub).toHaveBeenCalledTimes(1);
+        expect(comp.isAssessor).toBe(true);
+        expect(comp.complaint).not.toBe(null);
         fixture.detectChanges();
 
         const complaintsForm = debugElement.query(By.css('jhi-complaints-for-tutor-form'));
-        expect(complaintsForm).to.exist;
-        expect(comp.complaint).to.exist;
+        expect(complaintsForm).not.toBe(null);
+        expect(comp.complaint).not.toBe(null);
 
         // Wait until periodic timer has passed out
         tick(100);
+        flush();
     }));
 
     it('should lock a new submission', fakeAsync(() => {
         const activatedRoute: ActivatedRoute = fixture.debugElement.injector.get(ActivatedRoute);
         activatedRoute.params = of({ submissionId: 'new' });
         TestBed.inject(ActivatedRoute);
 
-        getProgrammingSubmissionForExerciseWithoutAssessmentStub.returns(of(submission));
+        getProgrammingSubmissionForExerciseWithoutAssessmentStub.mockReturnValue(of(submission));
 
         comp.ngOnInit();
         tick(100);
-        expect(getProgrammingSubmissionForExerciseWithoutAssessmentStub).to.be.calledOnce;
+        expect(getProgrammingSubmissionForExerciseWithoutAssessmentStub).toHaveBeenCalledTimes(1);
+        flush();
     }));
 
     it('should not show complaint when participation contains no complaint', fakeAsync(() => {
-        findBySubmissionIdStub.returns(of({ body: undefined }));
+        findBySubmissionIdStub.mockReturnValue(of({ body: undefined }));
         comp.ngOnInit();
         tick(100);
 
-        expect(getIdentityStub.calledOnce).to.be.true;
-        expect(lockAndGetProgrammingSubmissionParticipationStub.calledOnce).to.be.true;
-        expect(findBySubmissionIdStub.calledOnce).to.be.true;
-        expect(comp.complaint).to.not.exist;
+        expect(getIdentityStub).toHaveBeenCalledTimes(1);
+        expect(lockAndGetProgrammingSubmissionParticipationStub).toHaveBeenCalledTimes(1);
+        expect(findBySubmissionIdStub).toHaveBeenCalledTimes(1);
+        expect(comp.complaint).toBe(undefined);
         fixture.detectChanges();
 
         const complaintsForm = debugElement.query(By.css('jhi-complaints-for-tutor-form'));
-        expect(complaintsForm).to.not.exist;
+        expect(complaintsForm).toBe(null);
```

Comment:
[nit] `toBe(null)` or `toBe(undefined)`
I am unsure which one is our convention, but I guess it makes sense to stick with one approach.

Reply:
I had same the case in my PR and I learned that null is correct one: https://github.com/ls1intum/Artemis/pull/4104#discussion_r743950553

Reply:
we always prefer `undefined` over `null` in the client

Reply:
The [Sinon/Chai to Jest Conversion](https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=105252048) and the [Client Tests](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/) define the following: 
![Screenshot from 2021-11-07 09-36-07](https://user-images.githubusercontent.com/20031592/140636660-dc1c79c2-a370-4eac-ae0f-5b4890e6be5f.png)
![Screenshot from 2021-11-07 09-38-58](https://user-images.githubusercontent.com/20031592/140636663-c118012b-bf00-4d73-b8e4-846b4e121697.png)
Shall we change that? In the current case if I change `null` to `undefined` the test fails.

Reply:
if it is a CSS value, you can use toBe(null).